src/System.IO.Pipelines.Networking.Libuv/UvTcpConnection.cs

```diff
@@ -214,6 +223,8 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
                         }
                     }
                 }
+
+                inputBufferHandle.Free();
```

Comment:
Ok I think I get it now, I was conflating the lifetime of the handle with the lifetime of the buffer, rather than when it needs to be pinned.